### ISSUE TITLE
openssl_pkey_export(): cannot get key from parameter 1 

### ISSUE THREAD

**Issue by darkrain88** [nextcloud/server#11570]
### Steps to reproduce
1.new install NC 14.0.1 ()
2. 
3. i assume the problem maybe with the openssl version: 
root@Openwrt:/opt/wwwroot/default# opkg list-installed|grep ssl
libopenssl - 1.0.2p-1
php7-mod-openssl - 7.2.9-1
root@Openwrt:/opt/wwwroot/default# /bin/opkg list-installed|grep ssl
libevent2-openssl - 2.1.8-2
libopenssl - 1.0.2p-1
libustream-openssl - 2018-07-30-23a3f283-1
openssl-util - 1.0.2p-1
python-openssl - 2.7.15-1
transmission-daemon-openssl - 2.94-2
transmission-remote-openssl - 2.94-2
root@Openwrt:/opt/wwwroot/default#

### Expected behaviour

Tell us what should happen

login into server
`
### Actual behaviour
Tell us what happens instead

Internal Server Error
The server was unable to complete your request.

If this happens again, please send the technical details below to the server administrator.

More details can be found in the server log.
### Server configuration

Intel(R) Celeron(R) CPU J1900 @ 1.99GH

**Operating system**:
Linux Openwrt 4.14.70 #0 SMP Tue Sep 18 04:51:07 2018 x86_64
**Web server:**

nginx/1.14.0

**Database:**

mariadb-client - 10.2.17-1
mariadb-client-extra - 10.2.17-1
mariadb-server - 10.2.17-1
mariadb-server-extra - 10.2.17-1
nginx-extras - 1.14.0-1

**PHP version:**

php7 - 7.2.9-1
php7-cgi - 7.2.9-1
php7-cli - 7.2.9-1
php7-fastcgi - 7.2.9-1
php7-fpm - 7.2.9-1
php7-mod-calendar - 7.2.9-1
php7-mod-ctype - 7.2.9-1
php7-mod-curl - 7.2.9-1
php7-mod-dom - 7.2.9-1
php7-mod-exif - 7.2.9-1
php7-mod-fileinfo - 7.2.9-1
php7-mod-ftp - 7.2.9-1
php7-mod-gd - 7.2.9-1
php7-mod-gettext - 7.2.9-1
php7-mod-gmp - 7.2.9-1
php7-mod-hash - 7.2.9-1
php7-mod-iconv - 7.2.9-1
php7-mod-intl - 7.2.9-1
php7-mod-json - 7.2.9-1
php7-mod-ldap - 7.2.9-1
php7-mod-mbstring - 7.2.9-1
php7-mod-mysqli - 7.2.9-1
php7-mod-mysqlnd - 7.2.9-1
php7-mod-opcache - 7.2.9-1
php7-mod-openssl - 7.2.9-1
php7-mod-pcntl - 7.2.9-1
php7-mod-pdo - 7.2.9-1
php7-mod-pdo-mysql - 7.2.9-1
php7-mod-phar - 7.2.9-1
php7-mod-session - 7.2.9-1
php7-mod-shmop - 7.2.9-1
php7-mod-simplexml - 7.2.9-1
php7-mod-snmp - 7.2.9-1
php7-mod-soap - 7.2.9-1
php7-mod-sockets - 7.2.9-1
php7-mod-sqlite3 - 7.2.9-1
php7-mod-sysvmsg - 7.2.9-1
php7-mod-sysvsem - 7.2.9-1
php7-mod-sysvshm - 7.2.9-1
php7-mod-tokenizer - 7.2.9-1
php7-mod-xml - 7.2.9-1
php7-mod-xmlreader - 7.2.9-1
php7-mod-xmlwriter - 7.2.9-1
php7-mod-zip - 7.2.9-1
php7-pecl-dio - 0.1.0-1
php7-pecl-http - 3.2.0-1
php7-pecl-libevent - 2017-03-24-5a726cee95ef9760fe81a60e595e87655f324594-1
php7-pecl-propro - 2.0.1-2
php7-pecl-raphf - 2.0.0-2
php7-pecl-redis - 4.1.1-1

**Nextcloud version:** (see Nextcloud admin page)
14.0.1
**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install/ updated from 13.0.6
**Where did you install Nextcloud from:**
https://download.nextcloud.com/server/releases/
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
``` can pass through login page, no information
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
`
root@Openwrt:/opt/wwwroot/Nextcloud# sudo -u nobody php-cli occ app:list
Enabled:
  - accessibility: 1.0.1
  - activity: 2.7.0
  - cloud_federation_api: 0.0.1
  - comments: 1.4.0
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - federation: 1.4.0
  - files: 1.9.0
  - files_pdfviewer: 1.3.2
  - files_sharing: 1.6.2
  - files_texteditor: 2.6.0
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - files_videoplayer: 1.3.0
  - firstrunwizard: 2.3.0
  - gallery: 18.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - nextcloud_announcements: 1.3.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - support: 1.0.0
  - survey_client: 1.2.0
  - systemtags: 1.4.0
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - workflowengine: 1.4.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_external
  - user_ldap
``
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

`{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "www.myhome:443"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "14.0.1.1",
        "overwrite.cli.url": "https:\/\/www.myhome:443",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false
    }
}
``
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no




Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
"userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/69.0.3497.100 Safari\/537.36","version":"14.0.1.1"
**Operating system:**
win 10
#### Web server error log
<details>
<summary>Web server error log</summary>

`2409:8920:8813:843:6502:8056:c23:cb76, server: localhost,, request: "POST /index.php HTTP/1.1", upstream: "fastcgi://unix:/opt/var/run/php7-fpm.sock", host: "myhome443"
2018/10/03 06:43:12 [error] 20048#0: *32 access forbidden by rule, client: 2409:8a1e:8f0c:b4a5:75ed:8857:29fd:c3e4, server: localhost,, request: "GET /data/htaccesstest.txt HTTP/1.1", host: "myhome:443"
2018/10/03 06:55:19 [notice] 15668#0: signal process started
``
Insert your webserver log here
```
`2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:17 +0800] "POST /ajax.php HTTP/1.1" 200 1410 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:17 +0800] "POST /ajax.php HTTP/1.1" 200 1507 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:18 +0800] "POST /navigation.php?ajax_request=1 HTTP/1.1" 200 2105 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:18 +0800] "GET /favicon.ico HTTP/1.1" 200 22486 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:18 +0800] "POST /ajax.php HTTP/1.1" 200 1401 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:18 +0800] "POST /version_check.php HTTP/1.1" 200 75 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:18 +0800] "POST /db_structure.php?ajax_request=1&favorite_table=1&sync_favorite_tables=1 HTTP/1.1" 200 154 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:20 +0800] "GET /server_databases.php?server=1&ajax_request=true&ajax_page_request=true&_nocache=1538519600018221547&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 5667 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:20 +0800] "GET /themes/pmahomme/img/b_deltbl.png HTTP/1.1" 200 577 "https://www.helanhuahai.top:18443/phpmyadmin.css.php?nocache=6146841867ltr&server=1" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:33:20 +0800] "GET /js/server_databases.js?v=4.8.3 HTTP/1.1" 200 1982 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:34:04 +0800] "POST /server_databases.php HTTP/1.1" 200 1811 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:34:05 +0800] "POST /navigation.php?ajax_request=1 HTTP/1.1" 200 2141 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:34:05 +0800] "GET /db_structure.php?db=nextcloud&ajax_request=true&ajax_page_request=true&_nocache=1538519644895644500&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 2799 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:34:05 +0800] "GET /navigation.php?ajax_request=1&aPath=cm9vdA%3D%3D.bmV4dGNsb3Vk&vPath=cm9vdA%3D%3D.bmV4dGNsb3Vk&pos=0&pos2_name=&pos2_value=&searchClause=&searchClause2=&_nocache=1538519645195976243&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 1423 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:34:21 +0800] "GET / HTTP/1.1" 200 2618 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:42:49 +0800] "GET /db_structure.php?server=1&db=nextcloud&ajax_request=true&ajax_page_request=true&_nocache=1538520168208200880&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 9894 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:43:11 +0800] "POST /index.php HTTP/1.1" 504 585 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8a1e:8f0c:b4a5:75ed:8857:29fd:c3e4 - - [03/Oct/2018:06:43:12 +0800] "GET /data/htaccesstest.txt HTTP/1.1" 403 169 "-" "GuzzleHttp/6.2.1 curl/7.61.0 PHP/7.2.9"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:43:12 +0800] "GET /favicon.ico HTTP/1.1" 200 2623 "https://www.helanhuahai.top:8443/index.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:47:35 +0800] "GET /db_structure.php?server=1&db=nextcloud&ajax_request=true&ajax_page_request=true&_nocache=1538520453758654264&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 14125 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:47:38 +0800] "GET /navigation.php?ajax_request=1&aPath=cm9vdA%3D%3D.dGVzdA%3D%3D&vPath=cm9vdA%3D%3D.dGVzdA%3D%3D&pos=0&pos2_name=&pos2_value=&searchClause=&searchClause2=&_nocache=1538520458676807058&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 1423 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:47:40 +0800] "GET /navigation.php?ajax_request=1&aPath=cm9vdA%3D%3D.cGhwbXlhZG1pbg%3D%3D&vPath=cm9vdA%3D%3D.cGhwbXlhZG1pbg%3D%3D&pos=0&pos2_name=&pos2_value=&searchClause=&searchClause2=&_nocache=1538520459989354814&token=EFieV.%7F%2Cj~RB8esm HTTP/1.1" 200 2684 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:53:53 +0800] "GET / HTTP/1.1" 302 5 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:14 +0800] "GET /login HTTP/1.1" 200 3934 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:14 +0800] "GET /js/core/merged-template-prepend.js?v=d9d10faa-0 HTTP/1.1" 200 42176 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:14 +0800] "GET /js/core/merged-share-backend.js?v=d9d10faa-0 HTTP/1.1" 200 26512 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:15 +0800] "GET /js/core/merged-login.js?v=d9d10faa-0 HTTP/1.1" 200 2282 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:17 +0800] "GET /apps/theming/js/theming?v=0 HTTP/1.1" 200 232 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:30 +0800] "GET /apps/theming/styles?v=0 HTTP/1.1" 200 352 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:54:31 +0800] "GET /cron.php HTTP/1.1" 200 51 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:57:09 +0800] "GET /login HTTP/1.1" 200 3933 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:57:13 +0800] "GET /cron.php HTTP/1.1" 200 51 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:06:57:32 +0800] "POST /login HTTP/1.1" 500 5427 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:11:32 +0800] "POST /index.php HTTP/1.1" 200 1538 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:11:42 +0800] "POST /index.php HTTP/1.1" 200 1538 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:11:44 +0800] "POST /index.php HTTP/1.1" 200 84 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:11:44 +0800] "GET /db_structure.php?server=1&db=nextcloud HTTP/1.1" 200 3450 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:12:34 +0800] "GET /index.php/csrftoken HTTP/1.1" 200 101 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:16:29 +0800] "GET /tz.php HTTP/1.1" 200 9349 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:24:58 +0800] "GET /tz.php HTTP/1.1" 200 9339 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:26:12 +0800] "POST /tz.php HTTP/1.1" 200 9367 "https://www.helanhuahai.top:18443/tz.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:27:34 +0800] "GET /index.php/csrftoken HTTP/1.1" 200 103 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:28:25 +0800] "GET /tz.php?act=phpinfo HTTP/1.1" 200 27975 "https://www.helanhuahai.top:18443/tz.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:32:44 +0800] "GET /index.php/settings/integrity/failed HTTP/1.1" 303 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:32:44 +0800] "GET /login?redirect_url=/index.php/settings/integrity/failed HTTP/1.1" 200 3958 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:32:45 +0800] "GET /apps/theming/js/theming?v=0 HTTP/1.1" 200 232 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:32:57 +0800] "POST /login?redirect_url=/index.php/settings/integrity/failed HTTP/1.1" 500 5427 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:33:18 +0800] "GET /settings/integrity/failed HTTP/1.1" 303 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
2409:8920:8813:843:6502:8056:c23:cb76 - - [03/Oct/2018:07:33:18 +0800] "GET /login?redirect_url=/settings/integrity/failed HTTP/1.1" 200 3954 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"``
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
`log `{"reqId":"hhufh4vPqxmy77KhUiEf","level":3,"time":"2018-10-**02T22:57:32+00:00","remoteAddr":"2409:8920:8813:843:6502:8056:c23:cb76","user":"admin","app":"PHP","method":"POST","url":"\/login","message":"openssl_pkey_export(): cannot get key from parameter 1 at **\/opt\/wwwroot\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php#297","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/69.0.3497.100 Safari\/537.36","version":"14.0.1.1"}****
{"reqId":"hhufh4vPqxmy77KhUiEf","level":3,"time":"2018-10-02T22:57:32+00:00","remoteAddr":"2409:8920:8813:843:6502:8056:c23:cb76","user":"admin","app":"index","method":"POST","url":"\/login","message":{"Exception":"TypeError","Message":"openssl_pkey_get_details() expects parameter 1 to be resource, boolean given","Code":0,"Trace":[{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","line":300,"function":"openssl_pkey_get_details","args":[false]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","line":69,"function":"newToken","class":"OC\\Authentication\\Token\\PublicKeyTokenProvider","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/Authentication\/Token\/Manager.php","line":68,"function":"generateToken","class":"OC\\Authentication\\Token\\PublicKeyTokenProvider","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/User\/Session.php","line":631,"function":"generateToken","class":"OC\\Authentication\\Token\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/opt\/wwwroot\/Nextcloud\/core\/Controller\/LoginController.php","line":322,"function":"createSessionToken","class":"OC\\User\\Session","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"tryLogin","class":"OC\\Core\\Controller\\LoginController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\LoginController"},"tryLogin"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\LoginController"},"tryLogin"]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OC\\Core\\Controller\\LoginController","tryLogin",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"core.login.tryLogin"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"core.login.tryLogin"}]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"core.login.tryLogin"}]},{"file":"\/opt\/wwwroot\/Nextcloud\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/login"]},{"file":"\/opt\/wwwroot\/Nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/opt\/wwwroot\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","Line":300,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/69.0.3497.100 Safari\/537.36","version":"14.0.1.1"}``
</details>

#### Browser log
<details>
<summary>Browser log</summary>

Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by darkrain88** [comment]
@nextcloud-bot 

i ready submit the error in that thread 

this new.issue is not the privilege one.

can.you help me out the probolem.


whatever new.install or.update from 13.0.6 report.thesame error.

**Comment by darkrain88** [comment]
@danielkesselberg  thanks this is.new issue

**Comment by cbrace** [comment]
@darkrain88 Since upgrading my server several weeks ago, I have had this issue as well and and I am desperately looking for a workaround, since reverting to v13.0.4 is not possible. Did you find any kind of solution?

**Comment by gcosta74** [comment]
I have the same issue, but only after upgrading from MacOS High Sierra to MacOS Mojave.

I know that Nextcloud is not supported for MacOS but it worked since Nextcloud 10 without any problems until now

**Comment by gcosta74** [comment]
This is my log-entry

`{"reqId":"kxzQE2MroLHPsWkCwYbo","level":3,"time":"2018-11-28T17:14:22+00:00","remoteAddr":"91.209.160.17","user":"Pinuccio","app":"PHP","method":"POST","url":"\/index.php\/login?user=Pinuccio","message":"openssl_pkey_export(): cannot get key from parameter 1 at \/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php#297","userAgent":"Mozilla\/5.0 (Windows NT 10.0; WOW64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/69.0.3497.92 Safari\/537.36","version":"14.0.3.0"}
{"reqId":"kxzQE2MroLHPsWkCwYbo","level":3,"time":"2018-11-28T17:14:22+00:00","remoteAddr":"91.209.160.17","user":"Pinuccio","app":"index","method":"POST","url":"\/index.php\/login?user=Pinuccio","message":{"Exception":"TypeError","Message":"openssl_pkey_get_details() expects parameter 1 to be resource, boolean given","Code":0,"Trace":[{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","line":300,"function":"openssl_pkey_get_details","args":[false]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","line":69,"function":"newToken","class":"OC\\Authentication\\Token\\PublicKeyTokenProvider","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Authentication\/Token\/Manager.php","line":68,"function":"generateToken","class":"OC\\Authentication\\Token\\PublicKeyTokenProvider","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/User\/Session.php","line":631,"function":"generateToken","class":"OC\\Authentication\\Token\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/core\/Controller\/LoginController.php","line":322,"function":"createSessionToken","class":"OC\\User\\Session","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"tryLogin","class":"OC\\Core\\Controller\\LoginController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\LoginController"},"tryLogin"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\LoginController"},"tryLogin"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OC\\Core\\Controller\\LoginController","tryLogin",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"core.login.tryLogin"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"core.login.tryLogin"}]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"core.login.tryLogin"}]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/login"]},{"file":"\/Library\/WebServer\/Documents\/Nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/Library\/WebServer\/Documents\/Nextcloud\/lib\/private\/Authentication\/Token\/PublicKeyTokenProvider.php","Line":300,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows NT 10.0; WOW64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/69.0.3497.92 Safari\/537.36","version":"14.0.3.0"}`

**Comment by dustinwilson** [comment]
I started having this problem oddly when updating to 15.0. I went digging through here and to @danielkesselberg's [suggestion](https://github.com/nextcloud/server/issues/11227#issuecomment-421548385) in #11227. I in addition var_dumped `$config` and found it to be looking for the openssl configuration at `/absolute/location/of/openssl.cnf`. Looking through my config file sure enough it was set to that in there:

```
'openssl' => array(
        'config' => '/absolute/location/of/openssl.cnf',
),
```

I changed it to:
'openssl' => array(
        'config' => '/etc/ssl/openssl.cnf',
),

I am not sure why this did not give me any trouble in v14. I haven't looked through my backups to see if it was even there. I am not sure if this is the same trouble you are having, but I'm leaving this here in case this helps anyone else.

I think NextCloud should log this issue better. The openssl config file's location being set to something that doesn't exist should just show an error before openssl_pkey_get_details throws a fit.

**Comment by danielkesselberg** [comment]
> The openssl config file's location being set to something that doesn't exist should just show an error before openssl_pkey_get_details throws a fit.

Yes but this code is called very often and io access is expensive. I would vote against a check at this place.

Just another note. When you have

```
'openssl' => array(
        'config' => '/absolute/location/of/openssl.cnf',
),
```

in your config i would assume that you copied `config.sample.php` to `config.php` which is something you should not do. 

https://github.com/nextcloud/server/blob/4830fb7fb9dc355566ede942785d00a60323c68a/config/config.sample.php#L4-L10

Usually `config.php` is generated during the setup and you add additional options when you need them. If you are using the default openssl from your distribution there is no need to specify `openssl` in `config.php`.

**Comment by dustinwilson** [comment]
> Yes but this code is called very often and io access is expensive. I would vote against a check at this place.
Understandable, but if there were a way for it to show up only when initially logging in and not when repeatedly checking later it would be helpful.

> in your config i would assume that you copied config.sample.php to config.php which is something you should not do.
I didn't when I originally installed Nextcloud a while back. I upgraded to 15.0, and suddenly got this error which if present before the upgrade would have caused the same error, so I haven't a clue why it was there. I can't see how the upgrade process would have caused it, either, so it's not an error on your end. Leaving it here just in case someone else runs into this error, anyway.

**Comment by danielkesselberg** [comment]
> I upgraded to 15.0, and suddenly got this error which if present before the upgrade would have caused the same error,

https://github.com/nextcloud/server/pull/11240 dont blame me :rofl:  (< 15 a custom openssl configuration was not loaded at `PublicKeyTokenProvider`).

**Comment by kesselb** [comment]
Fixed by https://github.com/nextcloud/server/pull/16495
